packages/embark-contracts-manager/src/index.js

```diff
@@ -123,7 +123,8 @@ class ContractsManager {
 
           self.events.request("blockchain:contract:create", {abi: contract.abiDefinition, address: contract.deployedAddress}, async (contractObj) => {
             try {
-              const value = req.body.value !== undefined ? web3.utils.toWei(req.body.value, 'ether') : 0;
+              let value = typeof req.body.value === "number" ? req.body.value.toString() : req.body.value;
```

Comment:
We should consider very large wei values coming in from the client, and should always convert them to a `BigNumber` before sending through to the API. `BigNumber`s can handle very large ether amounts. For example, if a user tries to send `10000000000000000000000` wei, this would be represented as `1e22` which would get `.toString()`ed here. Instead, maybe we should also include the changes needed in Cockpit to ensure we are always sending `BigNumber`s.

Reply:
That's a good point. At the moment, we actually only allow sending `Ether` as unit. So super large amounts are rather unlikely until the point we allow users to choose the unit.

Reply:
I actually think the values are sent "raw", ie not converted, and therefore use the default unit which is wei, unless there is something I am missing?

Reply:
For clarification - the units are sent raw, yes, but are being expected in the API call as ether, and then converted to wei:
https://github.com/embark-framework/embark/blob/bd602b45ee3051b4363031699349f70d1b7a9095/packages/embark-contracts-manager/src/index.js#L126
So this is not really enforced in any way except by the UI. I'll remove the changes requested flag.

Maybe we should use a unit selector in the UI instead?

Reply:
> Maybe we should use a unit selector in the UI instead?

Yes, that's also what we discussed back when the input for sending ETH was introduced. Just hasn't happened yet.